rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystem.java

```diff
@@ -172,14 +182,15 @@ public FileSystem getRemoteFileSystem()
     return fs;
   }
 
-  public static void setLocalBookKeeper(BookKeeperService.Iface bookKeeper, String statsMbeanSuffix)
+  public static void setLocalBookKeeper(BookKeeperService.Iface bookKeeper, String statsMbeanSuffix, Configuration configuration)
   {
+    checkState(!initialized.get(), "LocalBookKeeper should be set before opening up the Filesystem to clients");
```

Comment:
should we just remove this check and reinitialize stats with new bookkeeper?

Reply:
Though this check is safe with the way Rubix is used in Presto right now: `setLocalBookKeeper` is called before FS initialize; But for safety we can remove this check and always override stats from here.